subsys/settings/src/settings_line.c

```diff
@@ -226,7 +226,7 @@ static int settings_line_raw_read_until(off_t seek, char *out, size_t len_req,
 	size_t exp_size, read_size;
 	u8_t rbs = settings_io_cb.rwbs;
 	off_t off;
-	int rc;
+	int rc = 0;
```

Comment:
This fixes false-negative static code analyse issue.
This is not a real issue as call to this function with request `len_req==0` doesn't make sense.

Reply:
While it may not make sense normally, it is true that this code will behave in an unpredictable way if the caller provides `len_req = 0` for whatever reason (e.g. a bug in their code)- introducing a hard-to-trace intermittent failure and, given the right conditions, even allowing CI to pass incorrect implementation. Also, this is a clear violation of MISRA C.

p.s. The real reason I am trying to do something about this is that certain GCC versions (4.5.1 for one) gives a warning about this and this warning is not totally unfounded.

Reply:
^^ I mean better to check invalid parameter explicitly

Reply:
and return -EINVAL

Reply:
I also noted that return value could be 1 if can not to finnd until_char, and this is not described in API documentation.

Reply:
> ^^ I mean better to check invalid parameter explicitly
> and return -EINVAL

Definitely. For this PR, I just wanted to make it have well defined behaviour without introducing any significant changes.

If you are okay with explicit parameter checking (and the performance penalty associated with it), I will push a new commit for that.

Reply:
yes, pleas do it this way